### DISCUSSION TITLE
finalUrl does not match requestedUrl

### DISCUSSION THREAD

**Discussion by CruddyShad0w** [GoogleChrome/lighthouse-ci#832] [Q&A]
When running lighthouse ci on a local project the output report is stating a different `finalUrl` than the `requestedUrl`.

```
{
  "lighthouseVersion": "9.3.0",
  "requestedUrl": "http://localhost:4500/#/test",
  "finalUrl": "http://localhost:4500/",
}
```

This is causing the generated reports to be labeled in-properly.
Is there a reason that the report would generate a `finalUrl` that does not match the `requestedUrl`?

When looking at the report, the contents of the report pertain specifically to the page http://localhost:4500/#/test

**Comment by jaspritt-hora** [comment]
@CruddyShad0w - This is WAI and happening cause - the url you are generating the report for is using `/#` . Lighthouse ci doesn't take into consideration the `url hash`. 

In your case - 
The path for the url is - `/`. You are using `hash` to differentiate the pages.

I have a similar scenario, and have added the implementation (PR is merged, yet to released) to handle this exact scenario. You can follow the discussion [here](https://github.com/GoogleChrome/lighthouse-ci/discussions/834).
The said [implementation](https://github.com/GoogleChrome/lighthouse-ci/pull/835) for reference.

**Reply by jaspritt-hora** [reply]
@CruddyShad0w - The url hash parameter handling is released. Refer the release [details](https://github.com/GoogleChrome/lighthouse-ci/releases/tag/v0.10.0).

In lighthousesrc.js, update the following parameter -

`reportFilenamePattern` with the value - '%%HOSTNAME%%-%%PATHNAME%%-%%HASH%%-%%DATETIME%%.report.%%EXTENSION%%'

**Reply by CruddyShad0w** [reply]
This did not end up resolving my issue.

I'm using the following upload parameters.
```
upload: {
    target: 'lhci',
    token: xxxxxxxxxxxxxxxxxxxx,
    reportFilenamePattern: '%%HOSTNAME%%-%%PATHNAME%%-%%HASH%%-%%DATETIME%%.report.%%EXTENSION%%',
},
```

Im guessing your fix works for generating the report files locally, but not for uploading to the LHCI dashboard directly?
Im also seeing that the finalUrl in the `lhr-xxxxxxxxxxx.json` does not contain the hash, which is the root of the problem for my implementation still. 
```
{
  "lighthouseVersion": "9.3.0",
  "requestedUrl": "http://localhost:4500/#/test",
  "finalUrl": "http://localhost:4500/",
}
```

**Reply by jaspritt-hora** [reply]
@CruddyShad0w - You are correct! The implementation done, is only for `target` : `filesystem`. The hash implementation needs to handled separately. 
Additionally, `reportFilenamePattern` is applied only when using `target: 'filesystem'`. 

In addition to this, the PR written for url hash handles part of the problem we are facing. Please refer https://github.com/GoogleChrome/lighthouse-ci/discussions/834 for further discussions.
